### PR TITLE
Feature/fixwindowsmzidtestfail

### PR DESCRIPTION
**Pull request by mwalzer**
should fix windows mzid test fail

### CODE REVIEW

**Comment by jpfeuffer** [comment]
Since it's a windows fix and the deleted parts are obviously not needed:
+1 for a quick merge

Just as a note. There is another failing test in the same file:
´´´
    creating new temporary filename 'MzIdentMLFile_test_242.tmp' (line 242)
Error: Caught std::exception
 - Message: bad lexical cast: source type value could not be interpreted as target
: failed
´´´

**Comment by timosachsenberg** [comment]
hmm could also be a problem in MzIdentMLHandler. There are a lot of boost::lexical_cast to string (e.g. from bool or String)

**Comment by timosachsenberg** [comment]
```
              String idec(boost::lexical_cast<std::string>((String(jt->getMetaValue("target_decoy"))).hasSubstring("decoy")));
```
might be invalid if there is no target_decoy meta value

**Comment by mwalzer** [comment]
Yeah, this is most probably due to missing target_decoy meta.... I'll look into it. No quick merge though please, i'll  integrate this here.

**Comment by mwalzer** [comment]
Of course TOPP_IDFileConverter_9_out1 failed because the optional isDecoy element is now treated as such ...

**Comment by timosachsenberg** [comment]
Hi Mathias,
if you look at CDASH output you can see that also the " is missing.

**Comment by timosachsenberg** [comment]
http://cdash.openms.de/testDetails.php?test=29094014&build=154627

File: src/openms/source/FORMAT/HANDLERS/MzIdentMLHandler.cpp [hunk 1]
```diff
@@ -743,26 +743,30 @@ namespace OpenMS
...
-              String idec(boost::lexical_cast<std::string>((String(jt->getMetaValue("target_decoy"))).hasSubstring("decoy")));
+              String idec;
+              if (jt->metaValueExists("target_decoy"))
+              {
+                idec = String(boost::lexical_cast<std::string>((String(jt->getMetaValue("target_decoy"))).hasSubstring("decoy")));
+              }
 
               String e;
-              e += "\t<PeptideEvidence id=\"" + pevid + "\" peptide_ref=\"" + pepid + "\" dBSequence_ref=\"" + dBSequence_ref;
+              e += "\t<PeptideEvidence id=\"" + pevid + "\" peptide_ref=\"" + pepid + "\" dBSequence_ref=\"" + dBSequence_ref + "\"";
```
Comment (timosachsenberg) [inline src/openms/source/FORMAT/HANDLERS/MzIdentMLHandler.cpp]: I think the trailing "\" is not correct here. if idec.empty() the line ends with an open quote. Could you check this again and maybe also add a test if it did not show up with your recent change?

Comment (mwalzer) [inline src/openms/source/FORMAT/HANDLERS/MzIdentMLHandler.cpp]: No, as you can see, all tests passed, except the mac build stuff.
I changed escaping to be line complete in that scope, means where there is an opening "\"" there should be a closing "\"" in the same line.

Comment (timosachsenberg) [inline src/openms/source/FORMAT/HANDLERS/MzIdentMLHandler.cpp]: ok perfect

**Pull request merged**